elastic/beats
# [elastic-agent] initial instrumentation

## What does this PR do?

Add some initial tracing to elastic-agent. I basically just looked through the code and was looking at lowhanging fruit for wrapping http handlers and clients. I'd love to have some of the EA dev's point out places I've missed that I could add more tracing, and ways to trigger these traces to view them in the UI.

todo:
- more instrumenting
- tests
- how to configure apm output? currently apm-agent-go just using default values and sending to the apm-server integration.

## Why is it important?

This will help in troubleshooting issues as more users start running EA.

## Checklist




## How to test this PR locally

build elastic-agent off this branch, start it under fleet, add the APM integration. the agent should appear as an apm service.

## Related issues

closes https://github.com/elastic/beats/issues/28895

## Use cases

troubleshooting mis-behaving elastic-agents

## Screenshots

![image](https://user-images.githubusercontent.com/1398104/142402500-60bdbdef-f16d-424f-aaf9-326a6d1a72a1.png)
(traces are coming from a metricbeat grabbing proxied data from apm-server)
